File: src/Payments/entities/Facture.java

package Payments.entities;

import java.time.LocalDate;

public class Facture {
     private int idFacture; 
     private int idCommande;
      private String etatPayment;
              private String dateFacture;
              private String name;

    public Facture(int idFacture, int idCommande, String etatPayment, String dateFacture, String name) {
        this.idFacture = idFacture;
        this.idCommande = idCommande;
        this.etatPayment = etatPayment;
        this.dateFacture = dateFacture;
        this.name = name;
    }
     public Facture(String etatPayment, String dateFacture, String name) {
       
        this.etatPayment = etatPayment;
        this.dateFacture = dateFacture;
        this.name = name;
    }
     public Facture( ) {}

   

  

    public int getIdFacture() {
        return idFacture;
    }

    public int getIdCommande() {
        return idCommande;
    }

    public String getEtatPayment() {
        return etatPayment;
    }

    public String getDateFacture() {
        return dateFacture;
    }

    public String getName() {
        return name;
    }

    public void setIdFacture(int idFacture) {
        this.idFacture = idFacture;
    }

    public void setIdCommande(int idCommande) {
        this.idCommande = idCommande;
    }

    public void setEtatPayment(String etatPayment) {
        this.etatPayment = etatPayment;
    }

    public void setDateFacture(String dateFacture) {
        this.dateFacture = dateFacture;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "Facture{" + "idFacture=" + idFacture + ", idCommande=" + idCommande + ", etatPayment=" + etatPayment + ", dateFacture=" + dateFacture + ", name=" + name + '}';
    }
              
}

File: src/Entity/Facture.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Facture
 *
 * @ORM\Table(name="facture", indexes={@ORM\Index(name="idCommande", columns={"idCommande"})})
 * @ORM\Entity
 */
class Facture
{
    /**
     * @var int
     *
     * @ORM\Column(name="idFacture", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idfacture;

    /**
     * @var int
     *
     * @ORM\Column(name="etatPayment", type="integer", nullable=false)
     */
    private $etatpayment;

    /**
     * @var string
     *
     * @ORM\Column(name="dateFacture", type="string", length=255, nullable=false)
     */
    private $datefacture;

    /**
     * @var \Commande
     *
     * @ORM\ManyToOne(targetEntity="Commande")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="idCommande", referencedColumnName="idCommande")
     * })
     */
    private $idcommande;

    public function getIdfacture(): ?int
    {
        return $this->idfacture;
    }

    public function getEtatpayment(): ?int
    {
        return $this->etatpayment;
    }

    public function setEtatpayment(int $etatpayment): self
    {
        $this->etatpayment = $etatpayment;

        return $this;
    }

    public function getDatefacture(): ?string
    {
        return $this->datefacture;
    }

    public function setDatefacture(string $datefacture): self
    {
        $this->datefacture = $datefacture;

        return $this;
    }

    public function getIdcommande(): ?Commande
    {
        return $this->idcommande;
    }

    public function setIdcommande(?Commande $idcommande): self
    {
        $this->idcommande = $idcommande;

        return $this;
    }


}
